client/src/modules/list/components/Items/ItemsList.jsx

```diff
@@ -21,9 +26,49 @@ class ItemsList extends PureComponent {
   }
 
   showMore = () =>
-    this.setState(({ limit }) => ({ limit: limit + DISPLAY_LIMIT }));
+    this.setState(
+      ({ limit }) => ({ limit: limit + DISPLAY_LIMIT }),
+      this.handleLimitUpdate
+    );
+
+  showLess = () =>
+    this.setState({ limit: DISPLAY_LIMIT }, this.handleLimitUpdate);
+
+  handleDisableAnimations = item => () => {
+    if (item.animate) {
+      const {
+        disableItemAnimations,
+        match: {
+          params: { id: listId }
+        }
+      } = this.props;
+
+      disableItemAnimations({ itemId: item._id, listId });
+    }
+  };
 
-  showLess = () => this.setState({ limit: DISPLAY_LIMIT });
+  handleLimitUpdate = () => {
+    const {
+      archived: archivedItemsList,
+      done: doneItemsList,
+      match: {
+        params: { id: listId }
+      },
+      updateLimit
+    } = this.props;
+    const { limit: currentLimit } = this.state;
+    const limit = {};
+
+    if (archivedItemsList) {
+      limit.archivedLimit = currentLimit;
+    } else if (doneItemsList) {
+      limit.doneLimit = currentLimit;
+    } else {
```

Comment:
it is necessary because we want to update only one limit. Without else unhandledLimit will be always saved in limit object and pass to redux. There is no return in earlier if/else if statements.